From eff9644153727e9a889ec3115c567d467bbbb96d Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 3 Aug 2023 12:37:29 +0800
Subject: [PATCH 07/18] CryptoPkg/openssl: move compiler_flags to buildinf.c

Seems with openssl 3.0 this is used by multiple source files,
so we get duplicate symbol errors when linking.  Fix that by
moving compiler_flags from header file to a source file.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Cc: Jiewen Yao <jiewen.yao@intel.com>
Cc: Xiaoyu Lu <xiaoyu1.lu@intel.com>
Cc: Guomin Jiang <guomin.jiang@intel.com>
Reviewed-by: Jiewen Yao <jiewen.yao@intel.com>
Acked-by: Ard Biesheuvel <ardb@kernel.org>
Tested-by: Ard Biesheuvel <ardb@kernel.org>
Tested-by: Brian J. Johnson <brian.johnson@hpe.com>
Tested-by: Kenneth Lautner <klautner@microsoft.com>
(cherry picked from commit 3af00aec7f829d379b12e0f969796fcf739e11ac)
---
 CryptoPkg/Library/OpensslLib/OpensslLib.inf       | 1 +
 CryptoPkg/Library/OpensslLib/OpensslLibCrypto.inf | 1 +
 CryptoPkg/Library/OpensslLib/buildinf.c           | 4 ++++
 CryptoPkg/Library/OpensslLib/buildinf.h           | 2 +-
 4 files changed, 7 insertions(+), 1 deletion(-)
 create mode 100644 CryptoPkg/Library/OpensslLib/buildinf.c

diff --git a/CryptoPkg/Library/OpensslLib/OpensslLib.inf b/CryptoPkg/Library/OpensslLib/OpensslLib.inf
index d25506abb2..8934dee2ea 100644
--- a/CryptoPkg/Library/OpensslLib/OpensslLib.inf
+++ b/CryptoPkg/Library/OpensslLib/OpensslLib.inf
@@ -28,6 +28,7 @@
 # Autogenerated files list starts here
 # Autogenerated files list ends here
   buildinf.h
+  buildinf.c
   ossl_store.c
   rand_pool.c
 
diff --git a/CryptoPkg/Library/OpensslLib/OpensslLibCrypto.inf b/CryptoPkg/Library/OpensslLib/OpensslLibCrypto.inf
index 2ef05100fa..cf3c0cae6f 100644
--- a/CryptoPkg/Library/OpensslLib/OpensslLibCrypto.inf
+++ b/CryptoPkg/Library/OpensslLib/OpensslLibCrypto.inf
@@ -28,6 +28,7 @@
 # Autogenerated files list starts here
 # Autogenerated files list ends here
   buildinf.h
+  buildinf.c
   ossl_store.c
   rand_pool.c
 
diff --git a/CryptoPkg/Library/OpensslLib/buildinf.c b/CryptoPkg/Library/OpensslLib/buildinf.c
new file mode 100644
index 0000000000..cd7fe9322e
--- /dev/null
+++ b/CryptoPkg/Library/OpensslLib/buildinf.c
@@ -0,0 +1,4 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent */
+#include "buildinf.h"
+
+const char  *compiler_flags = "compiler: information not available from edk2";
diff --git a/CryptoPkg/Library/OpensslLib/buildinf.h b/CryptoPkg/Library/OpensslLib/buildinf.h
index 9042c7526f..7cf0eae008 100644
--- a/CryptoPkg/Library/OpensslLib/buildinf.h
+++ b/CryptoPkg/Library/OpensslLib/buildinf.h
@@ -1,4 +1,4 @@
 #define PLATFORM  "UEFI"
 #define DATE      "Fri Dec 22 01:23:45 PDT 2017"
 
-const char  *compiler_flags = "compiler: information not available from edk2";
+extern const char  *compiler_flags;
-- 
2.43.0

